config/m4/fuse3.m4

```diff
@@ -10,21 +10,36 @@ AC_ARG_WITH([fuse3],
             [Enable the use of FUSEv3 (default is guess).])],
             [], [with_fuse3=guess])
 
-AS_IF([test "x$with_fuse3" != xno],
+AC_ARG_WITH([fuse3-static],
+            [AS_HELP_STRING([--with-fuse3-static=(DIR)],
+            [Static link FUSEv3 library (default is no).])],
+            [], [with_fuse3_static=no])
+
+AS_IF([test "x$with_fuse3" != "xno" -o "x$with_fuse3_static" != "xno"],
       [
-       AS_IF([test "x$with_fuse3" = "xguess" \
-                -o "x$with_fuse3" = "xyes" \
-                -o "x$with_fuse3" = "x"],
-             [FUSE3_CPPFLAGS=$(pkg-config --cflags fuse3)
-              FUSE3_LIBS=$(pkg-config --libs fuse3)],
-             [FUSE3_CPPFLAGS="-I${with_fuse3}/include/fuse3"
-              FUSE3_LIBS="-L${with_fuse3}/lib -L${with_fuse3}/lib64"])
+       AS_IF([test "x$with_fuse3" != "xguess" \
+                -a "x$with_fuse3" != "xyes" \
+                -a "x$with_fuse3" != "x" \
+                -a "x$with_fuse3" != "xno" \
+                -o "x$with_fuse3_static" != "xyes" \
```

Comment:
maybe when user sets with_fuse3_static=value we also set internally with_fuse3=value; this way can reduce some of the condition checks below
ideally, will keep only the condition in line 52